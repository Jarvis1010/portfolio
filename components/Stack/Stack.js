import styled from "styled-components";
import {
  spacing4,
  spacing8,
  spacing16,
  spacing32,
  spacing64,
  spacing128
} from "../constants/styled-constants";

const gaps = {
  "0": "0",
  "4": spacing4,
  "8": spacing8,
  "16": spacing16,
  "24": "1.5rem",
  "32": spacing32,
  "64": spacing64,
  "128": spacing128,
  default: "0"
};

const Stack = styled.div`
  display: grid;
  grid-gap: ${props => gaps[props.gapSize] || gaps.default};
  grid-auto-columns: 100%;
`;

export default Stack;
