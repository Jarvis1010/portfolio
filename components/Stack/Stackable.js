import styled from "styled-components";

const type = (theme = {}) => {
  const defaultTheme = {
    mediumGrey: "lightgrey",
    darkGrey: "grey",
    black: "black",
    white: "white"
  };
  const finalTheme = { ...defaultTheme, ...theme };
  return {
    light: { "background-color": finalTheme.white, color: finalTheme.black },
    medium: {
      "background-color": finalTheme.mediumGrey,
      color: finalTheme.black
    },
    dark: { "background-color": finalTheme.darkGrey, color: finalTheme.white },
    transparent: { "background-color": "transparent", color: finalTheme.white },
    default: { "background-color": finalTheme.white, color: finalTheme.black }
  };
};

const Stack = styled.div`
  display: flex;
  width: 100%;
  height: 100%;
  ${props =>
    Object.entries(type(props.theme)[props.type] || type().default)
      .map(x => x.join(":"))
      .join(";")};
`;

export default Stack;
