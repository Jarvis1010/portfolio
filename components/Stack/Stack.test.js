import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import Stack from "./Stack";

describe("Stack Tests", () => {
  it("renders with proper gaps", () => {
    ["0", "4", "8", "16", "24", "32", "64", "128", null].forEach(gapSize => {
      const wrapper = shallow(<Stack gapSize={gapSize} />);
      expect(toJSON(wrapper)).toMatchSnapshot();
    });
  });
});
