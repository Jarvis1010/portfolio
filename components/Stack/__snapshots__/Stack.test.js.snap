// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stack Tests renders with proper gaps 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  display: grid;
  grid-gap: ",
          [Function],
          ";
  grid-auto-columns: 100%;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  gapSize="0"
/>
`;

exports[`Stack Tests renders with proper gaps 2`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  display: grid;
  grid-gap: ",
          [Function],
          ";
  grid-auto-columns: 100%;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  gapSize="4"
/>
`;

exports[`Stack Tests renders with proper gaps 3`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  display: grid;
  grid-gap: ",
          [Function],
          ";
  grid-auto-columns: 100%;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  gapSize="8"
/>
`;

exports[`Stack Tests renders with proper gaps 4`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  display: grid;
  grid-gap: ",
          [Function],
          ";
  grid-auto-columns: 100%;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  gapSize="16"
/>
`;

exports[`Stack Tests renders with proper gaps 5`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  display: grid;
  grid-gap: ",
          [Function],
          ";
  grid-auto-columns: 100%;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  gapSize="24"
/>
`;

exports[`Stack Tests renders with proper gaps 6`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  display: grid;
  grid-gap: ",
          [Function],
          ";
  grid-auto-columns: 100%;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  gapSize="32"
/>
`;

exports[`Stack Tests renders with proper gaps 7`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  display: grid;
  grid-gap: ",
          [Function],
          ";
  grid-auto-columns: 100%;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  gapSize="64"
/>
`;

exports[`Stack Tests renders with proper gaps 8`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  display: grid;
  grid-gap: ",
          [Function],
          ";
  grid-auto-columns: 100%;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  gapSize="128"
/>
`;

exports[`Stack Tests renders with proper gaps 9`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  display: grid;
  grid-gap: ",
          [Function],
          ";
  grid-auto-columns: 100%;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  gapSize={null}
/>
`;
